swap/src/recover.rs

```diff
@@ -134,10 +134,87 @@ pub async fn alice_recover(
                 }
             };
         }
-        Alice::BtcRedeemable { redeem_tx, .. } => {
-            bitcoin_wallet
-                .broadcast_signed_transaction(redeem_tx)
-                .await?;
+        Alice::BtcRedeemable { redeem_tx, state } => {
+            let tx_lock_height = bitcoin_wallet
+                .transaction_block_height(state.tx_lock.txid())
+                .await;
+
+            let block_height = bitcoin_wallet.0.block_height().await?;
+            let refund_absolute_expiry = tx_lock_height + state.refund_timelock;
+
+            // bob cannot cancel
+            if block_height > refund_absolute_expiry {
```

Comment:
📣 Always tricky with these things, but I think it's this way round.

```suggestion
            if block_height < refund_absolute_expiry {
```